src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.IO;
+using System.Text;
+using Microsoft.AspNet.Mvc.Internal;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Wraps the supplied <see cref="Stream"/> to prevent it from flushing (which can cause response to be sent
+    /// in Chunked encoding) and disposing (as the hosting layers own the stream).
+    /// Also wraps the supplied <see cref="Encoding"/> to prevent writing the preamble or BOM bytes to the response.
+    /// </summary>
+    public class HttpResponseStreamWriter : StreamWriter
+    {
+        private const int DefaultBufferSize = 1024;
+
+        public HttpResponseStreamWriter(Stream stream, Encoding encoding)
+            : this(stream, encoding, DefaultBufferSize)
+        {
+        }
+
+        public HttpResponseStreamWriter(Stream stream, Encoding encoding, int bufferSize)
+            : base(new NonDisposableStream(stream), new NonBomEncodingWrapper(encoding), bufferSize, leaveOpen: true)
```

Comment:
Right, the `NonDisposable` part of the stream should not be a concern here as we will always leave the stream open and it is the flushing part only that we are interested...just wanted to reuse whatever we already have...